src/app/inventory/selectors.ts

```diff
@@ -106,25 +109,83 @@ export const materialsSelector = (state: RootState) =>
 /** The actual raw profile response from the Bungie.net profile API */
 export const profileResponseSelector = (state: RootState) => state.inventory.profileResponse;
 
+const STORE_SPECIFIC_OWNERSHIP_BUCKETS = [
+  // Emblems cannot be transferred between characters and if one character owns an emblem,
+  // other characters don't really own it. Also affects vendor claimability.
+  BucketHashes.Emblems,
+  // Quests and bounties are character-specific.
+  BucketHashes.Quests,
+];
+
+/**
+ * For most items, ownership is simple -- if it's in any bucket, we own this item.
+ * For items in some buckets (like quests and emblems), there is no concept of
+ * account-wide ownership. This mostly affects vendor claimability.
+ */
+export interface OwnedItems {
+  accountWideOwned: Set<number>;
+  storeSpecificOwned: Map<string, Set<number>>;
```

Comment:
In general I prefer a "plain object" whenever the key is a string - I don't think Map adds a lot in that case and it is much harder to work with (can't use spread operator, etc).